prow/istio-presubmit.sh

```diff
@@ -42,5 +42,5 @@ if [[ -n $(git diff) ]]; then
 fi
 
 # Upload images - needed by the subsequent tests
-time ISTIO_DOCKER_HUB="gcr.io/istio-testing" make push HUB="gcr.io/istio-testing" TAG="${GIT_SHA}"
+time ISTIO_DOCKER_HUB="gcr.io/istio-testing" make push HUB="gcr.io/istio-testing" TAG="${GIT_SHA}" DOCKER_BUILD_VARIANTS="default distroless"
```

Comment:
Sorry our prow stuff is in a lot of flux... I think we don't use this anymore as of a few days. Previously this was ran and built/pushed images, then other tests were kicked off. Now each test builds its own images.

I *think* your test should build its distroless image properly, but I am not sure how we can test it other than just having it run in the CI.

@icygalz can you take a look? Specifically at whether this will properly build and use the right image for the `prow/e2e-simple-distroless`?